CMakeLists.txt

```diff
@@ -158,6 +158,11 @@ if(MSVC)
         RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/$<0:>
         LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/$<0:>
         ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/$<0:>)
+
+    target_compile_options(${PROJECT_NAME} PRIVATE "$<$<CONFIG:Release>:/Zi>")
+    target_link_options(${PROJECT_NAME} PRIVATE "$<$<CONFIG:Release>:/DEBUG:FULL>")
+    target_link_options(${PROJECT_NAME} PRIVATE "$<$<CONFIG:Release>:/OPT:REF>")
```

Comment:
There should be no need to have this as three lines with `target_link_options(${PROJECT_NAME} PRIVATE` repeated. `target_link_options` takes a list. Also you could probably even combine them into one `$<$<CONFIG:Release>:/DEBUG:FULL /OPT:REF /OPT:ICF>` (although I haven't tested that)